powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/controller/api/PowerAuthControllerTest.java

```diff
@@ -17,112 +17,1047 @@
+        config.initActivation(powerAuthClient);
+        final RemoveActivationRequest removeActivationRequest = new RemoveActivationRequest();
+        removeActivationRequest.setActivationId(config.getActivationId());
+        removeActivationRequest.setExternalUserId(null);
+
+        final RemoveActivationResponse removeActivationResponse = powerAuthClient.removeActivation(removeActivationRequest);
+        assertTrue(removeActivationResponse.isRemoved());
+        assertEquals(config.getActivationId(), removeActivationResponse.getActivationId());
+
+        final GetActivationStatusRequest activationStatusRequest = new GetActivationStatusRequest();
+        activationStatusRequest.setActivationId(config.getActivationId());
+
+        final GetActivationStatusResponse statusResponse = powerAuthClient.getActivationStatus(activationStatusRequest);
+        assertEquals(ActivationStatus.REMOVED, statusResponse.getActivationStatus());
+    }
+
+    /**
+     * Tests the pagination functionality in retrieving an activation list for a specific user.
+     * <p>
+     * The test executes the following steps:
+     * <ol>
+     *   <li>Prepares a base request for fetching user activations, using user and application IDs from the configuration.</li>
+     *   <li>Creates 10 new activations for the user to ensure multiple pages of data.</li>
+     *   <li>Fetches the first page of activations, specifying page number and size, and validates the number of activations returned.</li>
+     *   <li>Fetches the second page of activations with a different page number but same page size, again validating the number of activations returned.</li>
+     *   <li>Asserts that the activations on the two pages are not identical, verifying the functionality of pagination.</li>
+     *   <li>Removes the created activations to maintain test isolation and ensure clean up after test execution.</li>
+     * </ol>
+     * </p>
+     *
+     * @throws Exception if any error occurs during the test execution, the initiation of activations, the removal of activations, or if the assertions fail.
+     */
+    @Test
+    void testActivationListForUserPagination() throws Exception {
+        final GetActivationListForUserRequest baseRequest = new GetActivationListForUserRequest();
+        baseRequest.setUserId(PowerAuthControllerTestConfig.USER_ID);
+        baseRequest.setApplicationId(config.getApplicationId());
+        final List<String> activationIds = new ArrayList<>();
+
+        for (int i = 0; i < 10; i++) {
+            final InitActivationRequest initActivationRequest = new InitActivationRequest();
+            initActivationRequest.setApplicationId(config.getApplicationId());
+            initActivationRequest.setUserId(PowerAuthControllerTestConfig.USER_ID);
+            final InitActivationResponse initResponse = powerAuthClient.initActivation(initActivationRequest);
+            activationIds.add(initResponse.getActivationId());
```

Comment:
It could be nice to use `stream` -> `map`, but since we declare checked exceptions, it is not possible (or not easily at least) 😞